arch/simddetect.cpp

```diff
@@ -26,7 +26,7 @@
 #endif // x86 target
 
 #if defined(X86_BUILD)
-# if defined(__linux__) || defined(__MINGW32__)
+# if defined(__GNUC__) || defined(__MINGW32__)
```

Comment:
Testing `__GNUC__` is sufficient, as MinGW also uses gcc, so the test for `__MINGW32__` can be removed.